src/polycurve.cpp

```diff
@@ -1,118 +1,48 @@
-}
-
-PolyCurve::PolyCurve(const PolyCurve& poly_curve) : PolyCurve(poly_curve.curves_) {}
-
-void PolyCurve::insertAt(uint idx, std::shared_ptr<Curve>& curve)
-{
-  Point s_1, s_2, e_1, e_2;
-  std::tie(s_1, e_1) = curve->endPoints();
-  if (idx > 0) // check with curve before
-  {
-    std::tie(s_2, e_2) = curves_[idx - 1]->endPoints();
-
-    double s_e = (s_1 - e_2).norm();
-    double e_e = (e_1 - e_2).norm();
-
-    if (e_e < s_e) // we need to reverse the curve
-    {
-      curve->reverse();
-      std::tie(s_1, e_1) = curve->endPoints();
-    }
-
-    curve->manipulateControlPoint(0, (s_1 + e_2) / 2);
-    curves_[idx - 1]->manipulateControlPoint(curves_[idx - 1]->order(), (s_1 + e_2) / 2);
-  }
-  if (idx + 1 < size()) // check with curve after
-  {
-    std::tie(s_2, e_2) = curves_[idx]->endPoints();
+PolyCurve::PolyCurve(std::deque<Curve> curves) : curves_(std::move(curves)) {}
 
-    double e_s = (e_1 - s_2).norm();
-    double s_s = (s_1 - s_2).norm();
+void PolyCurve::insertAt(unsigned int idx, Curve curve) { curves_.emplace(curves_.begin() + idx, curve); }
 
-    if (s_s < e_s) // we ned to reverse the curve
-    {
-      curve->reverse();
-      std::tie(s_1, e_1) = curve->endPoints();
-    }
+void PolyCurve::insertFront(Curve curve) { curves_.emplace_front(curve); }
 
-    curve->manipulateControlPoint(curve->order(), (e_1 + s_2) / 2);
-    curves_[idx + 1]->manipulateControlPoint(0, (e_1 + s_2) / 2);
-  }
+void PolyCurve::insertBack(Curve curve) { curves_.emplace_back(curve); }
 
-  curves_.insert(curves_.begin() + idx, curve);
-}
-
-void PolyCurve::insertFront(std::shared_ptr<Curve>& curve) { insertAt(0, curve); }
-
-void PolyCurve::insertBack(std::shared_ptr<Curve>& curve) { insertAt(size(), curve); }
-
-void PolyCurve::removeAt(uint idx)
-{
-  if (idx == 0)
-    removeFirst();
-  else if (idx == size() - 1)
-    removeBack();
-  else
-  {
-    Point s_1, s_2, e_1, e_2;
-    std::tie(s_1, e_1) = curves_[idx - 1]->endPoints();
-    std::tie(s_2, e_2) = curves_[idx + 1]->endPoints();
-    curves_[idx - 1]->manipulateControlPoint(curves_[idx - 1]->order(), (e_1 + s_2) / 2);
-    curves_[idx + 1]->manipulateControlPoint(0, (e_1 + s_2) / 2);
-    curves_.erase(curves_.begin() + idx);
-  }
-}
+void PolyCurve::removeAt(unsigned int idx) { curves_.erase(curves_.begin() + idx); }
 
 void PolyCurve::removeFirst() { curves_.pop_front(); }
 
 void PolyCurve::removeBack() { curves_.pop_back(); }
 
-PolyCurve PolyCurve::subPolyCurve(uint idx_l, uint idx_r) const
-{
-  return PolyCurve(std::deque<std::shared_ptr<Curve>>(curves_.begin() + idx_l, curves_.begin() + idx_r));
-}
-
-uint PolyCurve::size() const { return static_cast<uint>(curves_.size()); }
+unsigned int PolyCurve::size() const { return static_cast<unsigned int>(curves_.size()); }
 
-uint PolyCurve::curveIdx(double t) const
+unsigned int PolyCurve::curveIdx(double t) const
 {
-  uint idx = static_cast<uint>(t);
+  unsigned int idx = static_cast<unsigned int>(t);
```

Comment:
warning: use auto when initializing with a cast to avoid duplicating the type name [modernize-use-auto]